chaos_genius/alerts/anomaly_alerts.py

```diff
@@ -231,14 +290,20 @@ def _save_nl_message_hourly_freq(self, anomaly_data: List[dict], kpi: Kpi):
             else:
                 point["series_type"] = OVERALL_KPI_SERIES_TYPE_REPR
 
-        for point in prev_day_data:
-            if point["data_datetime"].hour not in data.keys():
-                data[point["data_datetime"].hour] = []
-            data[point["data_datetime"].hour].append(point)
+        hourly_data = dict()
+        if time_series_freq in ("h", "H", "hourly", "Hourly"):
+            for point in prev_day_data:
+                if point["data_datetime"].hour not in hourly_data.keys():
+                    hourly_data[point["data_datetime"].hour] = []
+                hourly_data[point["data_datetime"].hour].append(point)
 
         for point in anomaly_data:
-            hour_val = point["data_datetime"].hour
-            intended_point = self._find_point(point, data.get(hour_val, []))
+            if time_series_freq in ("d", "D", "daily", "Daily"):
+                intended_point = self._find_point(point, prev_day_data)
+            elif time_series_freq in ("h", "H", "hourly", "Hourly"):
+                hour_val = point["data_datetime"].hour
+                intended_point = self._find_point(point, hourly_data.get(hour_val, []))
+            
             if intended_point is None:
```

Comment:
Also, would be better to separate this percent change calculation into a function so that this part just becomes:
```python
point["percentage_change"] = find_percentage_change(point["y"], intended_point["y"] if intended_point else None)
```